### PR TITLE
refactor(framework):EO-1147 Refactor Flexbox Grid

### PR DESCRIPTION
**Pull request by satyamkotakonda**
JIRA: : https://jira.rax.io/browse/EO-1147

### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM
- [x] Design LGTM
- ~~[ ] QE LGTM (no modification to functional javascript)~~

Before
--------

After
-------
screen shot 2016-09-19 at 3 31 04 pm

### CODE REVIEW

**Comment by satyamkotakonda** [comment]
@CITguy  Getting design issue.
Here is the folder structure  and design issue 





Please help me here.

File: src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+
+  <h2>Overview</h2>
+
+  <p>
+    Angular Material's responsive CSS layout is built on
+    <a href="http://www.w3.org/TR/css3-flexbox/" target=_blank>flexbox.</a>
+  </p>
+
+  <p>
+    The layout system is based upon element attributes rather than CSS classes. Attributes provide an easy way to set a value
```
Comment (sundip) [inline src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html]: This has more than 120 characters in the line.

File: src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+    <div flex class="flex-box light-blue">
+      [flex]
+    </div>
+    <div flex class="flex-box dark-green">
+      [flex]
+    </div>
+  </div>
+
+  <p>
+    Add the <code>flex</code> attribute to a layout's child element, and it will flex (stretch) to fill the available area.
```
Comment (sundip) [inline src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html]: Character count is more the 120

File: src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+
+  <p>
+    Add the <code>flex-order</code> attribute to a layout child to set its position within the layout.
+    Any value from 0-9 is accepted.
+  </p>
+
+  <p>
+    Note that the <code>flex-order</code> attribute is not compatible with the <code>layout-margin</code> attribute.
+    This is because the CSS selector engine selects based on DOM markup order and the <code>layout-margin</code>
+    attribute makes use of <code>:first-child</code> and <code>:last-child</code> to apply margins toonly the inner elements in
```
Comment (sundip) [inline src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html]: character count

File: src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html [hunk 4]
```diff
@@ -0,0 +1,554 @@
...
+      [flex offset="33"]
+    </div>
+    <div flex class="flex-box light-blue">
+      [flex]
+    </div>
+  </div>
+
+  <p>
+    Add the <code>offset</code> attribute to a layout child to set its offset percentage within the layout.
+    Values must be multiples of <code>5</code>, or <code>33</code>, <code>34</code>, <code>66</code>, <code>67</code>.
```
Comment (sundip) [inline src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html]: character count

File: src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html [hunk 5]
```diff
@@ -0,0 +1,554 @@
...
+
+  <p>
+    The <code>layout-align</code> attribute takes two parameters in any order. Parameters <code>top</code>,
+    <code>middle</code>, <code>bottom</code>, <code>left</code>, <code>right</code>, and <code>center</code> 
+    determine the alignment of child elements and may be combined in sensible ways (ie <code>top left</code> works but
+    <code>top bottom</code> would not).
+  </p>
+
+  <p>
+    Parameters <code>stretch</code>, <code>justify</code>, and <code>spread</code> determine the justification of child elements.
```
Comment (sundip) [inline src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html]: character count, this and the block above.

**Comment by satyamkotakonda** [comment]
@sundip  I will update and my assumption went wrong here.

File: src/utilities/FlexboxGrid/docs/flexboxGrid.docs.html [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  <a href="https://material.angularjs.org/#/layout/container">Angular Material's layout module</a>
+  with minor usability enhancements to provide an assortment of attribute-based layout options based on the flexbox layout model.
+  Included are intuitive attribute based styles that ease the creation of responsive row and/or column based page layouts.
+</p>
+
+<h3><rx-permalink>Layout</rx-permalink></h3>
+<rx-example name="flexbox.layout"></rx-example>
+
+<h3><rx-permalink>Flexbox Grid</rx-permalink></h3>
+<p>
```
Comment (KalpeshBhandari) [inline src/utilities/FlexboxGrid/docs/flexboxGrid.docs.html]: @satyamkotakonda This para reflects two times.

File: src/utilities/FlexboxGrid/docs/flexboxGrid.docs.html [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  Encore UI includes a grid system forked from 
+  <a href="https://material.angularjs.org/#/layout/container">Angular Material's layout module</a>
+  with minor usability enhancements to provide an assortment of attribute-based layout options based on the flexbox layout model.
```
Comment (KalpeshBhandari) [inline src/utilities/FlexboxGrid/docs/flexboxGrid.docs.html]: @satyamkotakonda There is extra space at end of line.

File: src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html [hunk 8]
```diff
@@ -0,0 +1,554 @@
...
+    <a href="http://www.w3.org/TR/css3-flexbox/" target=_blank>flexbox.</a>
+  </p>
+
+  <p>
+    The layout system is based upon element attributes rather than CSS classes. Attributes provide an easy way to set a value
+    (eg <code>layout="row"</code>) and help separate concerns: attributes define layout, and classes define styling.
+  </p>
+
+  <!-- Layout Attribute -->
+  <a id="Attribute"></a>
```
Comment (KalpeshBhandari) [inline src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html]: @satyamkotakonda @CITguy Is this still useful? There are some more instances too.

File: src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html [hunk 9]
```diff
@@ -0,0 +1,554 @@
...
+      <td>flex-order-lg</td>
+      <td>Sets element order on devices between 960px and 1200px wide.</td>
+    </tr>
+    <tr>
+      <td>flex-order-gt-lg</td>
+      <td>Sets element order on devices greater than 1200px wide.</td>
+    </tr>
+  </table>
+
+
```
Comment (KalpeshBhandari) [inline src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html]: @satyamkotakonda Extra line break

File: src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html [hunk 10]
```diff
@@ -0,0 +1,554 @@
...
+      <td>flex-lg</td>
+      <td>Sets flex on devices between 960px and 1200px.</td>
+    </tr>
+    <tr>
+      <td>flex-gt-lg</td>
+      <td>Sets flex on devices greater than 1200px wide.</td>
+    </tr>
+  </table>
+
+
```
Comment (KalpeshBhandari) [inline src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html]: @satyamkotakonda Extra line break

File: src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html (hunk 8 above)
Comment (CITguy) [inline src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html]: No. Please remove this line.

File: src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html [hunk 11]
```diff
@@ -0,0 +1,554 @@
+<div class="layout-demo">
```
Comment (CITguy) [inline src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html]: This should not be in an example. 

Please migrate content to `src/utilities/FlexboxGrid/docs/FlexboxGrid.docs.html`.

File: src/utilities/FlexboxGrid/docs/flexboxGrid.docs.html [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+</p>
+
+<h3><rx-permalink>Layout</rx-permalink></h3>
+<rx-example name="flexbox.layout"></rx-example>
+
+<h3><rx-permalink>Flexbox Grid</rx-permalink></h3>
+<p>
+  Encore UI includes a grid system forked from 
+  <a href="https://material.angularjs.org/#/layout/container">Angular Material's layout module</a>
+  with minor usability enhancements to provide an assortment of attribute-based layout options based on the flexbox layout model.
```
Comment (CITguy) [inline src/utilities/FlexboxGrid/docs/flexboxGrid.docs.html]: Please remove this paragraph.

File: src/utilities/FlexboxGrid/docs/flexboxGrid.docs.html [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  Included are intuitive attribute based styles that ease the creation of responsive row and/or column based page layouts.
+</p>
+<p>
+  Angular Material's responsive CSS layout is built on <a href="http://www.w3.org/TR/css3-flexbox/" target="_blank">flexbox.</a>
+  The layout system is based upon element attributes rather than CSS classes.
+  Attributes provide an easy way to set a value (eg <code>layout="row"</code>) and help separate concerns:
```
Comment (CITguy) [inline src/utilities/FlexboxGrid/docs/flexboxGrid.docs.html]: Replace top paragraph in file with this paragraph.

File: src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html (hunk 2 above)
Comment (satyamkotakonda) [inline src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html]: screen shot 2016-09-15 at 3 59 37 pm

File: src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html [hunk 14]
```diff
@@ -0,0 +1,554 @@
...
+  </p>
+
+  <p>
+    The <code>flex</code> attribute value is restricted to 33, 66, and multiples of five.
+    For example: <code>flex="5"</code>, <code>flex="20"</code>, "<code>flex="33"</code>, <code>flex="50"</code>,
+    <code>flex="66"</code>, <code>flex="75"</code>, ....
+  </p>
+
+  <p>
+    See the <a href="#/components/layout#Options">layout options</a> for more information on responsive flex attributes.
```
Comment (satyamkotakonda) [inline src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html]: screen shot 2016-09-15 at 4 01 10 pm

File: src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html (hunk 4 above)
Comment (satyamkotakonda) [inline src/utilities/FlexboxGrid/docs/examples/flexbox.layout.html]: screen shot 2016-09-15 at 4 04 26 pm

**Comment by satyamkotakonda** [comment]
@KalpeshBhandari  @sundip  @CITguy  Updated changes and please review.

**Comment by satyamkotakonda** [comment]
@CITguy  Thanks for suggestions and  design issue got resolved.

File: src/utilities/FlexboxGrid/docs/flexboxGrid.docs.html [hunk 15]
```diff
@@ -0,0 +1,558 @@
...
+  attributes define layout,and classes define styling.
+</p>
+
+<h3><rx-permalink>Layout</rx-permalink></h3>
```
Comment (CITguy) [inline src/utilities/FlexboxGrid/docs/flexboxGrid.docs.html]: Please remove lines 1-8.

screen shot 2016-09-15 at 9 57 46 am

File: src/utilities/FlexboxGrid/flexboxGrid.meta.json [hunk 16]
```diff
@@ -0,0 +1,13 @@
+{
+    "displayName": "FlexboxGrid",
```
Comment (CITguy) [inline src/utilities/FlexboxGrid/flexboxGrid.meta.json]: Please add space between "Flexbox" and "Grid" so that the display name is `"Flexbox Grid"`

**Comment by CITguy** [comment]
Dev LGTM

@satyamkotakonda: please update screenshots for Design review.

**Comment by satyamkotakonda** [comment]
@CITguy Screenshots updated.
Thanks.

**Comment by cgtaab** [comment]
Design LGTM

**Pull request merged**